ui/job-view/headerbars/SecondaryNavBar.jsx

```diff
@@ -0,0 +1,296 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+
+import treeherder from '../../js/treeherder';
+import { thEvents } from '../../js/constants';
+import { getBtnClass } from '../../helpers/job';
+import { getUrlParam } from '../../helpers/location';
+import WatchedRepo from './WatchedRepo';
+
+class SecondaryNavBar extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { $injector } = this.props;
+    this.ThResultSetStore = $injector.get('ThResultSetStore');
+    this.ThRepositoryModel = $injector.get('ThRepositoryModel');
+    this.thJobFilters = $injector.get('thJobFilters');
+    this.$rootScope = $injector.get('$rootScope');
+    this.$location = $injector.get('$location');
+
+    this.filterChicklets = [
+      'failures',
+      this.thJobFilters.filterGroups.nonfailures.resultStatuses,
+      'in progress'].reduce((acc, val) => acc.concat(val), []);
+    const searchStr = this.thJobFilters.getFieldFiltersObj().searchStr;
+    this.repoName = getUrlParam('repo');
+
+    this.state = {
+      groupsExpanded: getUrlParam('group_state') === 'expanded',
+      showDuplicateJobs: getUrlParam('duplicate_jobs') === 'visible',
+      resultStatusFilters: this.thJobFilters.getResultStatusArray(),
+      searchQueryStr: searchStr ? searchStr.join(' ') : '',
+      watchedRepos: [],
+      allUnclassifiedFailureCount: 0,
+      filteredUnclassifiedFailureCount: 0,
+    };
+  }
+
+  componentDidMount() {
+    this.toggleGroupState = this.toggleGroupState.bind(this);
+    this.toggleFieldFilterVisible = this.toggleFieldFilterVisible.bind(this);
+    this.toggleUnclassifiedFailures = this.toggleUnclassifiedFailures.bind(this);
+    this.clearFilterBox = this.clearFilterBox.bind(this);
+    this.unwatchRepo = this.unwatchRepo.bind(this);
+
+    this.unlistenGlobalFilterChanged = this.$rootScope.$on(thEvents.globalFilterChanged, () => {
+      this.updateToggleFilters();
+      this.setState({ searchQueryStr: this.getSearchStr() });
+    });
+    this.unlistenRepositoriesLoaded = this.$rootScope.$on(thEvents.repositoriesLoaded, () => (
+      this.setState({ watchedRepos: this.ThRepositoryModel.watchedRepos })
+    ));
+    this.unlistenJobsLoaded = this.$rootScope.$on(thEvents.jobsLoaded, () => (
+      this.setState({
+        allUnclassifiedFailureCount: this.ThResultSetStore.getAllUnclassifiedFailureCount(),
+        filteredUnclassifiedFailureCount: this.ThResultSetStore.getFilteredUnclassifiedFailureCount(),
+      })
+    ));
+  }
+
+  componentWillUnmount() {
+    this.unlistenGlobalFilterChanged();
+    this.unlistenRepositoriesLoaded();
+    this.unlistenJobsLoaded();
+  }
+
+  getSearchStr() {
+    const ss = this.thJobFilters.getFieldFiltersObj().searchStr;
```

Comment:
ss is short for searchString presumably?

Reply:
It is, but probably better to spell it out for clarity.